plugins/MauticTagManagerBundle/Views/Tag/details.html.twig

```diff
@@ -0,0 +1,71 @@
+{% extends 'MauticCoreBundle:Default:content.html.twig' %}
+
+{% block mauticContent %}tagmanager{% endblock %}
+{% block headerTitle %} {{ tag.getTag() }} {% endblock %}
+
+{% set customButtons = {} %}
+
+{% block actions %}
+    {{- include('MauticCoreBundle:Helper:page_actions.html.twig', {
+        item : tag,
+        nameGetter: 'getTag',
+        templateButtons: {
+            edit: security.isGranted('tagManager:tagManager:edit'),
+            delete:  security.isGranted('tagManager:tagManager:delete'),
+            close:  security.isGranted('tagManager:tagManager:edit'),
+        },
+        routeBase: 'tagmanager'
+    }) -}}
+{% endblock %}
+
+{% block content %}
+    <!-- start: box layout -->
+    <div class="box-layout">
+        <!-- left section -->
+        <div class="col-md-12 bg-white height-auto">
+            <div class="bg-auto">
+                <!-- page detail header -->
+                <!-- sms detail collapseable toggler -->
+                <div class="pr-md pl-md pt-lg pb-lg">
+                    <div class="box-layout">
+
+                        <div class="col-xs-10">
+                            <div class="text-white dark-sm mb-0">{{  tag.getDescription()|raw }}</div>
```

Comment:
One big perk for switching to TWIG is that we'll get rid of many XSS holes where unescaped PHP can contain JS and other harmful code. Using the `|raw` filter means we go back to unescaped code. I searched the internets and some people recommend

https://github.com/Exercise/HTMLPurifierBundle

which can play nicely with Twig. Or can we use https://twig.symfony.com/doc/3.x/filters/escape.html? I don't fully understand what it does from the docs.

Now the question is whether to deal with security before we convert everything to Twig within this refactoring of if we'll do it afterwards as not part of this refactoring. @RCheesley what's your take on that?

Nitpick: Can you please standardize the spaces inside the Twig tokens? I can see that they are using 1 space after `{{` and one space before `}}` in the [docs](https://twig.symfony.com/) so let's use that format everywhere.
```suggestion
                            <div class="text-white dark-sm mb-0">{{ tag.getDescription()|raw }}</div>
```

Reply:
Twig doesn't provide the option to prevent XSS,

also `escape` will escape all HTML tags in its way.

we have to use one of the following solutions:

- use [HTMLPurifierBundle](https://github.com/Exercise/HTMLPurifierBundle)
- create our filter
- use [striptags](https://twig.symfony.com/doc/3.x/filters/striptags.html) but this solution isn't recommended as it'll require a lot of work to give it every HTML tag we need to be whitelisted


In the meantime, I added TODOs in the twig template to be easier to get back to it later

Reply:
I checked what Drupal recommends. [They just state](https://www.drupal.org/docs/security-in-drupal/writing-secure-code-for-drupal#s-sanitizing-on-output-to-avoid-cross-site-scripting-xss-attacks)

>  watch out for the "raw" filter, which does not escape output. Only use this when you are certain the data is trusted.

which is not really helpful. We need to support some basic HTML in the output but not anything harmful and we cannot trust the data as it's user input.

I found this XSS filter method from Drupal 8 but it looks quite simple to solve the big problem like XSS sanitization:

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Xss.php/function/Xss%3A%3Afilter/8.2.x

So I'd recommend to install https://github.com/Exercise/HTMLPurifierBundle and use that.

Reply:
Yeah, the best two options we have is to install the bundle or create the filter,

I'll be more than happy to implement any of them once approved.

Reply:
Ok. let's implement https://github.com/Exercise/HTMLPurifierBundle in this PR and merge this.